letsencrypt-auto-source/pieces/fetch.py

```diff
@@ -49,14 +50,14 @@ def __init__(self):
             if isinstance(handler, HTTPHandler):
                 self._opener.handlers.remove(handler)
 
-    def get(self, url):
+    def get(self, url, timeout=socket._GLOBAL_DEFAULT_TIMEOUT):
         """Return the document contents pointed to by an HTTPS URL.
 
         If something goes wrong (404, timeout, etc.), raise ExpectedError.
 
         """
         try:
-            return self._opener.open(url).read()
+            return self._opener.open(url, timeout = timeout).read()
         except (HTTPError, IOError) as exc:
```

Comment:
I think you should add `urllib2.URLError` here.

Reply:
Suggesting because, when I do `.open(url, timeout=some_super_short_timeout)` it throws a URLError.

Reply:
I agree. I just got pulled away from keyboard before pushing.

URLError is a subclass of IOError; Are you concerned about an uncaught exception or is the idea to simply catalogue all the likely exception classes?

Reply:
If it's a subclass, we should be all set.